lib/trino-parquet/src/main/java/io/trino/parquet/reader/SimpleSliceInputStream.java

```diff
@@ -113,4 +120,46 @@ public int readIntUnsafe()
         offset += Integer.BYTES;
         return value;
     }
+
+    /**
+     * Use this method only if there is a significant performance boost validated in benchmarks.
```

Comment:
I would just leave the other part of sentence:
```
     * Always check if needed data is available with ensureBytesAvailable method.
     * Failing to do so may result in instant JVM crash.
```
Either this method is dangerous or not